src/capture.rs

```diff
@@ -0,0 +1,75 @@
+use crate::Activated;
+use crate::Error;
+use crate::Packet;
+use crate::State;
+use crate::{raw, Capture, PacketHeader};
+use libc::c_int;
+use libc::c_uchar;
+
+/// The type for the function to pass into pcap_loop
+pub type HandlerFunc = fn(Packet) -> ();
+
+/// An internal function to convert the callback capture into rust safe function,
+/// #Safety
+/// using mem::transmute here is fine since `HandlerFunc` **Should** be the equivalent of
+/// pcap_handler (removing the extern and unsafe keyword, need to check if removing them have
+/// unidentified behavior although we are converting from and into the same type)
+unsafe extern "C" fn capturer(
+    params: *mut c_uchar,
+    raw_header: *const raw::pcap_pkthdr,
+    raw_pkt_data: *const c_uchar,
+) {
+    let callback: HandlerFunc = std::mem::transmute(params);
+    let header = *(raw_header as *const PacketHeader);
```

Comment:
is raw_header the same structure as PacketHeader?

Reply:
You can see this also in [pcap_next](https://github.com/ItamarElkarif/pcap/blob/master/src/lib.rs#L1074)
and it works (assuming since the structs are with the same structure and size so the primitive cast works